.github/workflows/build.yaml

```diff
@@ -496,16 +517,22 @@ jobs:
       run: make mock-grpc-server-image
 
     - name: Set up Docker Buildx
+      # Skip for external contributions.
+      if: github.event.pull_request.head.repo.full_name == github.repository
       uses: docker/setup-buildx-action@v2
 
     - name: Login to quay.io/rhacs-eng
+      # Skip for external contributions.
+      if: github.event.pull_request.head.repo.full_name == github.repository
       uses: docker/login-action@v2
       with:
         registry: quay.io
         username: ${{ secrets.QUAY_RHACS_ENG_RW_USERNAME }}
         password: ${{ secrets.QUAY_RHACS_ENG_RW_PASSWORD }}
 
     - name: Push to quay.io/rhacs-eng
+      # Skip for external contributions.
+      if: github.event.pull_request.head.repo.full_name == github.repository
```

Comment:
How about for `build-and-push` action just skip `push` so we can check if image is buildable? Just like`dex` do it but with different condition.

https://github.com/dexidp/dex/blob/master/.github/workflows/docker.yaml#L81